### ISSUE TITLE
Use Location Service always

### ISSUE THREAD

**Issue by powerlocke-ctrl** [cgeo/cgeo#15193]
### Tell us your idea!

Hi guys!
I am using a smart watch to get an Message, when  I arrived an area near to a cache.
Because c:geo uses only Location Service, when display is on and in foreground, I think it could be helpful to give c:geo permission for using location always. Also if c:geo is running in background.

### What solution would you suggest?

permission for using location always

### Is there an alternative?

Put the screen on and use c:geo in foreground only

**Comment by fm-sys** [comment]
Our toggle talking works while screen is off since PR https://github.com/cgeo/cgeo/pull/13085. In short, foreground services can access the GPS location, even if c:geo is running in background. Something like this can be adopted for "nearby cache" notifications as well.

**Comment by powerlocke-ctrl** [comment]
![Screenshot_20240114-233011_Berechtigungssteuerung](https://github.com/cgeo/cgeo/assets/58145821/92d61d3b-6249-4c8e-8589-6216a97a36aa)
![Screenshot_20240114-233035_Berechtigungssteuerung](https://github.com/cgeo/cgeo/assets/58145821/eecc5141-4dc6-46d7-b840-2711f6a6a845)

I am not sure, if it’s clear, what I want to describe or ask for.
I’ve attached two screenshots, one from google maps, where the location permission is settleable to always. The other shows same permission for c:geo, where you see, permission is not settlabe to always.

Always, when my display is turned on after a while off, c:geo needs a few seconds to find the actual position.
That’s the reason, why I cannot get any notification to my watch or on mobile phone, when the screen is off.

**Comment by Lineflyer** [comment]
> I’ve attached two screenshots, one from google maps, where the location permission is settleable to always. The other shows same permission for c:geo, where you see, permission is not settlabe to always.

Requesting the permanent permission to access the location is nothing which is just a quick switch AFAIR. 
Google requires apps, which want to use that to justify the reason and require further assessment and screening. Nothing what can be easily done and should be well considered, whether its worth it.

**Comment by fm-sys** [comment]
I think cgeo as Geocaching app which highly relies on GPS would have really good chances to get passed the Google Play checks for background location access.

That being said, I have to take away the illusion that everything will instantly get better with just requesting the permission. In fact, the foreground permission is enough for all our cases. As of now, it's just a lack of implementation.

To prove my point, I can recommend [this GPS tracker app](https://play.google.com/store/apps/details?id=com.twoscape.sportler). It completely works with foreground location permission, however it manages to record the GPS track over hours even if screen is off or app is running in background.

**Comment by moving-bits** [comment]
A more detailed description for expected behavior was given by @anondev-sudo in https://github.com/cgeo/cgeo/issues/15533#issue-2215894874:

When the option to play sound for closest or most distant geocaches or waypoints is enabled in the settings, proximity notifications do not trigger when the phone screen is turned off. This results in missed geocaches as notifications are not received while the phone is in the pocket.

Steps to Reproduce:
Enable the option to play sound for closest or most distant geocaches or waypoints.
Turn off the phone screen.
Approach a geocache or waypoint to trigger proximity notifications.
Note that no notifications are received while the screen is off.

Expected Behavior:
Proximity notifications should still trigger and play sound even when the phone screen is turned off, ensuring users are alerted to nearby geocaches or waypoints.
Actual Behavior:
Notifications are not received when the phone screen is off, leading to missed geocaches and the need to backtrack.

**Comment by moving-bits** [comment]
see similar request by @MagpieFourtyTwo (https://github.com/cgeo/cgeo/issues/15708#issue-2284387060)

### Tell us your idea!

When caching by bike, you also have to struggle with the phone's battery, obviously.
And as display is *the* main battery eater, best bet is to switch if off as soon (and long) as possible.
But with display off, it's kind of hard to not miss the next cache.  Or stage.
Being the smart kid that I am, thought, just activate proximity notifications and you're on the safe side.
Big miss.  No signal at all.  Of course missed the next one and had to head back to get it.  :(  ;)

### What solution would you suggest?

Keep track of GPS position even if display is off (others do so anyway, thus energy needs should be negligible) and let alarms ring, even with display off,

Moreover I would suggest to play ``Specific waypoint proximity notification``, too, if you approach current routing target resp. next routing point (like navigation devices do for decades ;) ).

### Is there an alternative?

_No response_

### Additional Information

_No response_

**Comment by anondev-sudo** [comment]
Yes I agree with this. And I think that there should be also option to turn off sound notifications on caches with found it logs, and some stages/waypoints.

**Comment by anondev-sudo** [comment]
When will be this fixed? I would be useful if you go on powertrails that you can hear voice notification that there is a cache nearby and you should start looking at it. It would save time, because I often miss geocaches when I hike and don't look that often on my phone,, and have to return back

**Comment by moving-bits** [comment]
> When will be this fixed? 

As soon as some volunteer finds the time, has the knowledge and is interested in implementing this...

Please keep in mind, that all work on c:geo is done by volunteers in their spare time, there's no company backing this work.

Pull requests from contributors are very welcomed, though.

**Comment by ririko5834** [comment]
Let's hope someone will find time to implement this feature so we don't have to look at the phone screen often while hiking on geocaching trails to avoid missing a cache and having to return back
